lib/src/rules/always_specify_types.dart

```diff
@@ -149,9 +179,25 @@ class _Visitor extends SimpleAstVisitor<void> {
         param.type == null &&
         !isJustUnderscores(identifier.name)) {
       if (param.keyword != null) {
-        rule.reportLintForToken(param.keyword);
-      } else {
-        rule.reportLint(param);
+        var keyword = param.keyword!;
+        var type = param.declaredElement?.type;
+
+        if (keyword.type == Keyword.VAR &&
+            type != null &&
+            type is! DynamicType) {
+          rule.reportLintForToken(keyword,
+              arguments: [keyword, type], errorCode: keywordCouldBeTypeCode);
+        } else {
+          rule.reportLintForToken(keyword);
+        }
+      } else if (param.declaredElement != null) {
+        var type = param.declaredElement!.type;
+
+        if (type is DynamicType) {
+          rule.reportLint(param);
```

Comment:
Do we want to use a specific errorCode here?

Reply:
Related to the below case, I am not sure which specific `errorCode` to be used.

```dart
b(s) {}
```

Reply:
Ok, we can leave it for now.